src/main/java/com/eprosima/fastdds/idl/templates/TypeObjectTestingTestSource.stg

```diff
@@ -0,0 +1,1323 @@
+$get_type_identifier(type=typedefs.typedefContentTypeCode, var="related_type_ids")$
+EXPECT_TRUE(related_type_ids.type_identifier1() == type_objects.minimal_type_object.alias_type().body().common().related_type() ||
+    related_type_ids.type_identifier2() == type_objects.minimal_type_object.alias_type().body().common().related_type());
+EXPECT_TRUE(related_type_ids.type_identifier1() == type_objects.complete_type_object.alias_type().body().common().related_type() ||
+    related_type_ids.type_identifier2() == type_objects.complete_type_object.alias_type().body().common().related_type());
+$check_member_detail_annotations(member=typedefs, type="alias_type().body()", parent=typedefs)$
+>>
+
+test_structure(struct) ::= <<
+TEST(TypeObjectTests, TestStructureTypeObject_$struct.name$)
+{
+    register_$ctx.filename$_type_objects();
+
+    $check_struct_type(struct=struct)$ 
+}
+>>
+
+check_struct_type(struct) ::= <<
+ReturnCode_t ret_code;
+TypeIdentifierPair type_ids;
+$get_type_identifier_registry(typename=struct.scopedname, result="type_ids")$
+$check_direct_hash_type_identifier(typeid="type_ids")$
+TypeObjectPair type_objects;
+$get_type_object_registry(typename=struct.scopedname, result="type_objects")$
+EXPECT_EQ(TK_STRUCTURE, type_objects.minimal_type_object._d());
+EXPECT_EQ(TK_STRUCTURE, type_objects.complete_type_object._d());
+$expected_type_flags(object=struct)$
+EXPECT_EQ($struct.name$_expected_flags, type_objects.minimal_type_object.struct_type().struct_flags());
+EXPECT_EQ($struct.name$_expected_flags, type_objects.complete_type_object.struct_type().struct_flags());
+$if(struct.inheritance)$
+TypeIdentifierPair base_type_ids;
+$get_type_identifier_registry(typename=struct.inheritance.scopedname, result="base_type_ids")$
+$check_direct_hash_type_identifier(typeid="base_type_ids")$
+EXPECT_TRUE(base_type_ids.type_identifier1() == type_objects.minimal_type_object.struct_type().header().base_type() ||
+    base_type_ids.type_identifier1() == type_objects.complete_type_object.struct_type().header().base_type());
+EXPECT_TRUE(base_type_ids.type_identifier2() == type_objects.minimal_type_object.struct_type().header().base_type() ||
+    base_type_ids.type_identifier2() == type_objects.complete_type_object.struct_type().header().base_type());
+TypeObjectPair base_type_objects;
+$get_type_object_registry(typename=struct.inheritance.scopedname, result="base_type_objects")$
+$else$
+TypeIdentifier invalid_type_id;
+EXPECT_EQ(invalid_type_id, type_objects.minimal_type_object.struct_type().header().base_type());
+EXPECT_EQ(invalid_type_id, type_objects.complete_type_object.struct_type().header().base_type());
+$endif$
+$check_type_detail_annotations(object=struct, type="struct_type().header().detail()")$
+EXPECT_EQ("$struct.scopedname$", type_objects.complete_type_object.struct_type().header().detail().type_name().to_string());
+$if (struct.members)$
+MemberId member_id = 0;
+$if (struct.inheritance)$
+    member_id = $struct.firstMember.id$;
```

Comment:
This specific case cannot be generated because the TypeObject reorders the elements and the specification states that the id should be next one after the last id of the inherited structure. But it is not possible to know which is the last element because the elements are reordered in the TypeObject. Thus, for this specific case, the id given by IDL Parser API is used.